internal/template/template.go

```diff
@@ -128,6 +133,12 @@ func MergeData(dst map[string]interface{}, src map[string]interface{}) error {
 	return mergo.Merge(&dst, src, mergoOpts...)
 }
 
+// CopyData creates a deep copy of the argument map src.
+func CopyData(src map[string]interface{}) map[string]interface{} {
+	data := copier.Copy(src).(map[string]interface{})
```

Comment:
Since you are doing a type assertion here, we probably should also catch errors if the type of return value is not map[string]interface{} , e.g.

data, ok = ...
if not ok {
  # log error
}
return data